code/gcy/gcy_model.py

```diff
@@ -58,16 +62,98 @@ def __init__(self,
                     ρ_zπ=0.970,          # ρ_hxπ
                     s_zπ=0.271):         # σ_hxπ
 
-        self.β, self.ψ, self.γ = β, ψ, γ 
-        self.ρ_λ, self.s_λ, self.μ_c, self.φ_c, self.ρ = ρ_λ, s_λ, μ_c, φ_c, ρ 
+        self.β, self.ψ, self.γ = β, ψ, γ
+        self.ρ_λ, self.s_λ, self.μ_c, self.φ_c, self.ρ = ρ_λ, s_λ, μ_c, φ_c, ρ
         self.ρ_π, self.φ_z, self.ρ_c = ρ_π, φ_z, ρ_c
-        self.s_c, self.ρ_z, self.s_z = s_c, ρ_z, s_z 
+        self.s_c, self.ρ_z, self.s_z = s_c, ρ_z, s_z
         self.ρ_ππ, self.φ_zπ, self.ρ_zπ, self.s_zπ = ρ_ππ, φ_zπ, ρ_zπ, s_zπ
 
         # Pack params into an array
-        self.params = (β, ψ, γ, 
-                         ρ_λ, s_λ, μ_c, φ_c, ρ, 
-                         ρ_π, φ_z, ρ_c, s_c, ρ_z, s_z, 
+        self.params = (β, ψ, γ,
+                         ρ_λ, s_λ, μ_c, φ_c, ρ,
+                         ρ_π, φ_z, ρ_c, s_c, ρ_z, s_z,
                          ρ_ππ, φ_zπ, ρ_zπ, s_zπ)
 
 
+## ==== Log linear approximation of the GCY W/C ratio === ##
+
+def wc_loglinear_factory(gcy):
+    """
+    A factory function that computes the constant terms for the WC ratio log
+    linear approximation and then creates a jitted function that evaluates the
+    log linear approximation.
+
+    """
+
+    # Unpack parameters
+    (β, ψ, γ,
+        ρ_λ, s_λ, μ_c, φ_c, ρ,
+        ρ_π, φ_z, ρ_c, s_c, ρ_z, s_z,
+        ρ_ππ, φ_zπ, ρ_zπ, s_zπ) = gcy.params
+    θ = (1 - γ) / (1 - 1/ψ)
+
+    s_wc = 2*(φ_c)**2*s_c
+    s_wx = 2*(φ_z)**2*s_z
+    s_wxπ = 2*(φ_zπ)**2*s_zπ
+
+    def fk1(x):
+        return np.exp(x)/(1+np.exp(x))
+
+    def fk0(x):
+        return np.log(1+np.exp(x))-fk1(x)*x
+
+    def fA1(x):
+        return (1-1/ψ)/(1-fk1(x)*ρ)
+
+    def fAλ(x):
+        return ρ_λ/(1-fk1(x)*ρ_λ)
+
+    def fAπ(x):
+        return fk1(x)*(1-1/ψ)*ρ_π/((1-fk1(x)*ρ)*(1-fk1(x)*ρ_ππ))
+
+    def fAz(x):
+        return (θ/2)*(fk1(x)*fA1(x))**2/(1-fk1(x)*ρ_z)
+
+    def fAzπ(x):
+        return (θ/2)*(fk1(x)*fAπ(x))**2/(1-fk1(x)*ρ_zπ)
+
+    def fAc(x):
+        return (θ/2)*(1-1/ψ)**2/(1-fk1(x)*ρ_c)
+
+    def fA0(x):
+        return (np.log(β)+fk0(x)+μ_c * (1-1/ψ) \
+                + fk1(x) * fAz(x) * (φ_z)**2 * (1-ρ_z)\
+                + fk1(x) * fAc(x) * (φ_c)**2 * (1-ρ_c)\
+                + fk1(x) * fAzπ(x) * (φ_zπ)**2 * (1-ρ_zπ)\
+                + (θ/2) * ((fk1(x) * fAλ(x)+1)**2 * s_λ**2 \
+                + (fk1(x) * fAz(x) * s_wx)**2+(fk1(x) * fAc(x) * s_wc)**2\
+                + (fk1(x) * fAzπ(x) * s_wxπ)**2)) \
+                / (1-fk1(x))
+
+    def fq_bar(x):
+        return x - fA0(x) - fAc(x)*(φ_c)**2 - fAz(x)*(φ_z)**2 - fAzπ(x)*(φ_zπ)**2
+
+    qbar = brentq(fq_bar, -20, 20)
```

Comment:
I tested this by plotting the graph and the root is: `6.061309329603693` at default GCY params